src/jmath/rand.ts

```diff
@@ -87,8 +87,8 @@ export function chooseObjectWithProbability<T extends objectWithProbability>(
     (maxProbability, current) => current.probability + maxProbability,
     0,
   );
-  // Choose random integer within the sum of all the probabilities
-  const roll = randInt(1, maxProbability, seedRandomInstance);
+  // Choose random number within the sum of all the probabilities
+  const roll = randFloat(0, maxProbability, seedRandomInstance);
```

Comment:
This is a significant change.
Why float instead of int?
With 0 as the lower bound this would take 1-2 (a 50/50 chance) and change it to 0-2 (33/33/33).
With a change like this I'd expect to see an explanation in the PR comments and added tests to cover the new functionality.
My intuition is to keep this as it was, but I'm curious about your reasoning here.

Reply:
Rand float will give the same result as rand int currently does, but will also handle non-integer probabilities if given

For the example you gave, the chance is still 50/50. Two things with a probability of 1 would generate a number between 0 (inclusive) and 2 (exclusive). Anything less than 1 = result of 1, anything less than 2 = result of 2. Both have an equal number of possibilities (0-0.999 and 1-1.999)

I did a little more research to double check before responding, and I also found that [Math.Random() has an exclusive maximum](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random), so I may update the surrounding code to reflect that as well (mostly comments and variable names, change <= to < on line 57, and remove the probability != 0 exception I added on line 54)

other source: [Will JavaScript random function ever return a 0 or 1](https://stackoverflow.com/questions/11730263/will-javascript-random-function-ever-return-a-0-or-1)